consensus/core/src/broadcaster.rs

```diff
@@ -0,0 +1,227 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    cmp::{max, min},
+    sync::Arc,
+    time::Duration,
+};
+
+use consensus_config::AuthorityIndex;
+use futures::{stream::FuturesUnordered, StreamExt as _};
+use tokio::{
+    sync::broadcast,
+    task::JoinSet,
+    time::{error::Elapsed, sleep_until, timeout, Instant},
+};
+use tracing::{trace, warn};
+
+use crate::{
+    block::VerifiedBlock, context::Context, core::CoreSignalsReceivers, error::ConsensusResult,
+    network::NetworkClient,
+};
+
+/// Number of Blocks that can be inflight sending to a peer.
+const BROADCAST_CONCURRENCY: usize = 10;
```

Comment:
My understanding is blocks are produced every ~400ms, if concurrency of 10 will mostly be reached only when a client is fairly unresponsive, might that higher than usual concurrency be worsening any congestion related issue? Very naive take here, just having a poke around.

Reply:
Block creation interval is about the broadcast latency to reach a quorum of validators, usually 80ms ~ 120ms. Each block should have size up to 100KB usually. So even with 10 inflight, it should be a minor fraction of bandwidth on most validators.